### ISSUE TITLE
Restore cursor position

### ISSUE THREAD

**Issue by astier** [numToStr/Comment.nvim#29]
Currently when the cursor is in the middle of a paragraph and I comment it out with `gcap` the cursor is moved to the beginning of the paragraph. I would like the cursor not to move but to stay at the same position.

With vim-commentary I worked around that by overriding `gcap` with a mapping which would create a mark before commenting the paragraph and than move the cursor back to that mark.

```vim
nn gcap my<cmd>norm vip<bar>gc<cr>`y
```

However, because of issue #24 the mark gets deleted after commenting with Comment.nvim and the cursor can not jump back.

**Comment by astier** [comment]
I actually think that a user would expect such behavior. The only reason I see why users wouldn't expect it is because they are already used and conditioned to this after years of using comment-plugins which move the cursor-position.

I also think making this behavior default would be more consistent with how `gcc` in Comment.nvim currently behaves. `gcc` also doesn't move the cursor position to the beginning of the line whereas vim-commentary does move the cursor to the beginning of the line.

It's your decision after all but I personally wouldn't be to afraid to break from old norms if the new one is better.

Counter question. Are there reasons to move the cursor? It always felt arbitrary and disruptive to me.

**Comment by numToStr** [comment]
> It's your decision after all but I personally wouldn't be to afraid to break from old norms if the new one is better.

Same feelings 😂. As I said earlier, I also want this behaviour. So, I'll implement this sooner or later.

> Are there reasons to move the cursor? It always felt arbitrary and disruptive to me.

IDK this might be a vim thingy. I didn't added any special cursor handling as of now. And most of the mappings like `gcc` `gc10j` doesn't move the cursor position.

**Comment by numToStr** [comment]
Try getting cursor position inside pre and post hooks. That should clear things up a bit.

**Comment by jandamm** [comment]
This observation should be true.
The opfunc is called with the current selection applied. So after the cursor has changed.
Only way I'd see is to use a mark.
For example prefix `i<ESC>` to set the mark and then after the opfunc jump to mark `.`.

**Comment by astier** [comment]
I just checked it for myself and understand now what you mean @numToStr If at
https://github.com/numToStr/Comment.nvim/blob/a86384dc153e86a9a3cbec1436a8acfa4b07690d/lua/Comment/comment.lua#L169
the cursor-position is already moved before the function is called because as @jandamm said at this point the selection has already be done and the cursor moved than the position has to be stored before the selection is done.

I just tried @jandamm PR and it works for me. It solves it more or less how I imagined it. I am not sure if using marks is a good solution. Wouldn't this potentially pollute/overwrite the user-marks?
